packages/animations/src/animation_builder.ts

```diff
@@ -83,3 +88,146 @@ export abstract class AnimationFactory {
+                  'or by importing \`BrowserAnimationsModule\`/\`NoopAnimationsModule\` in your application');
+    }
+  }
+
+  override build(animation: AnimationMetadata|AnimationMetadata[]): AnimationFactory {
+    const id = this._nextAnimationId;
+    this._nextAnimationId++;
+    const entry = Array.isArray(animation) ? sequence(animation) : animation;
+    issueAnimationCommand(this._renderer, null, id, 'register', [entry]);
+    return new BrowserAnimationFactory(id, this._renderer);
+  }
+}
+
+class BrowserAnimationFactory extends AnimationFactory {
+  constructor(private _id: number, private _renderer: Renderer2) {
+    super();
+  }
+
+  override create(element: any, options?: AnimationOptions): AnimationPlayer {
+    return new RendererAnimationPlayer(this._id, element, options || {}, this._renderer);
+  }
+}
+
+class RendererAnimationPlayer implements AnimationPlayer {
+  public parentPlayer: AnimationPlayer|null = null;
+  private _started = false;
+
+  constructor(
+      public id: number, public element: any, options: AnimationOptions,
+      private _renderer: Renderer2) {
+    this._command('create', options);
+  }
+
+  private _listen(eventName: string, callback: (event: any) => any): () => void {
+    return this._renderer.listen(this.element, `@@${this.id}:${eventName}`, callback);
+  }
+
+  private _command(command: string, ...args: any[]): void {
+    issueAnimationCommand(this._renderer, this.element, this.id, command, args);
+  }
+
+  onDone(fn: () => void): void {
+    this._listen('done', fn);
+  }
+
+  onStart(fn: () => void): void {
+    this._listen('start', fn);
+  }
+
+  onDestroy(fn: () => void): void {
+    this._listen('destroy', fn);
+  }
+
+  init(): void {
+    this._command('init');
+  }
+
+  hasStarted(): boolean {
+    return this._started;
+  }
+
+  play(): void {
+    this._command('play');
+    this._started = true;
+  }
+
+  pause(): void {
+    this._command('pause');
+  }
+
+  restart(): void {
+    this._command('restart');
+  }
+
+  finish(): void {
+    this._command('finish');
+  }
+
+  destroy(): void {
+    this._command('destroy');
+  }
+
+  reset(): void {
+    this._command('reset');
+    this._started = false;
+  }
+
+  setPosition(p: number): void {
+    this._command('setPosition', p);
+  }
+
+  getPosition(): number {
+    return unwrapAnimationRenderer(this._renderer)?.engine?.players[this.id]?.getPosition() ?? 0;
+  }
+
+  public totalTime = 0;
+}
+
+function issueAnimationCommand(
+    renderer: Renderer2, element: any, id: number, command: string, args: any[]): void {
+  renderer.setProperty(element, `@@${id}:${command}`, args);
+}
+
+/**
+ * The following 2 methods cannot reference their correct types (AnimationRenderer &
+ * DynamicDelegationRenderer) since this would introduce a import cycle.
+ */
+
+function unwrapAnimationRenderer(renderer: Renderer2): {engine: {players: AnimationPlayer[]}}|null {
+  if ((renderer as any).ɵtype === RendererTypes.Animation) {
```

Comment:
I'd propose avoiding `as any` and type-cast instead, for example:
```suggestion
  const type = (renderer as {ɵtype: RendererTypes}).ɵtype;
  if (type === RendererTypes.Animation) {
```